flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -389,11 +389,20 @@ public Task(
 			++counter;
 		}
 
+		invokableHasBeenCanceled = new AtomicBoolean(false);
+
+		// finally, create the executing thread, but do not start it
+		executingThread = new Thread(TASK_THREADS_GROUP, this, taskNameWithSubtask);
+
+		// add metrics for buffers
+		this.metrics.getIOMetricGroup().initializeBufferMetrics(this);
+
 		// register detailed network metrics, if configured
 		if (tmConfig.getBoolean(TaskManagerOptions.NETWORK_DETAILED_METRICS_KEY)) {
-			MetricGroup networkGroup = metricGroup.addGroup("Network"); // same as in MetricUtils.instantiateNetworkMetrics()
-			MetricGroup outputGroup = networkGroup.addGroup("Output"); // this is optional
-			MetricGroup inputGroup = networkGroup.addGroup("Input"); // this is optional
+			// similar to MetricUtils.instantiateNetworkMetrics() but inside this IOMetricGroup
+			MetricGroup networkGroup = this.metrics.getIOMetricGroup().addGroup("Network");
```

Comment:
this is equivalent to ```metricGroup.addGroup("Network")```. The IOMetricGroup is just a proxy that forwards calls to the parent group, i.e. the task metric group.